woocommerce/sensei-certificates
# Add View Certificate block.

Fixes #261

### Changes proposed in this Pull Request
- Adding new `sensei-certificate/view-certificate` block.
- This block relies on existing method to generate the certificate link.
- Logic to generate the certificate link has been updated to:
  - Hide block if user didn't complete the course yet.
  - Get course_id from current course as fallback if GET is not set.
- Added `@wordpress/icon` dependency to set the "button" icon for the
  new block. **Not sure why is this needed and not the other wordpress dependencies** ?


### Testing instructions
* Create a Course.
* Choose a Certificate Template on the right sidebar.
* Add a View Certificate block using the slash inserter or the inserter.
* Check that button does not appear if the user didn't complete the course yet.
* Check that button appears when user completes course.
* Confirm that the certificate is shown correctly on click.


### Screenshot / Video
#### Editor
![image](https://user-images.githubusercontent.com/799065/177552314-d33dcb46-9586-44d5-abf8-c1be5dae5f4c.png)

#### Student's view
![image](https://user-images.githubusercontent.com/799065/177552548-87528fd1-fac9-46ed-be44-640d1abaca17.png)


### Discussion / Extra details
- This was more confusing than I anticipated.
- There was already a `sensei-certificates/view-certificate-button` block which is a variation from `core/button`.
  - The problem with this variation is that `core/button` can only be used in `core/buttons` and it can't be displayed in the slash inserter (more info [here](p1657100493834919-slack-C45SNKV4Z)).
  - This variation is registered to the Course Completed Actions block, which is in Sensei but this block currently works only on pages (expected to be used in the Completed Course separate page that can be setup in the settings). Making this to work in Courses would be quite easy, but decided to descope that for the moment since I don't think it is being used very much.
- The new block uses the same mechanism that was being used to generate the certificate link for the Course Completed Actions block but I had to modify it slightly to support the current course id and to hide the button if the course is not complete.
